crates/provider-sdk/src/provider.rs

```diff
@@ -499,33 +506,47 @@ impl ProviderConnection {
         P: Provider,
     {
-        match deserialize::<InterfaceLinkDefinition>(&msg.payload) {
-            Ok(ld) => {
-                let span = tracing::Span::current();
-                span.record("source_id", &tracing::field::display(&ld.source_id));
-                span.record("target", &tracing::field::display(&ld.target));
-                span.record("wit_namespace", &tracing::field::display(&ld.wit_namespace));
-                span.record("wit_package", &tracing::field::display(&ld.wit_package));
-                span.record(
-                    "wit_interfaces",
-                    &tracing::field::display(&ld.interfaces.join(",")),
-                );
-                span.record("link_name", &tracing::field::display(&ld.name));
-                // If the link has already been put, return early
-                if self.is_linked(&ld.source_id).await {
-                    warn!("Ignoring duplicate link put");
-                    return;
-                }
-
-                info!("Linking actor with provider");
-                if provider.put_link(&ld).await {
-                    self.put_link(ld).await;
-                } else {
-                    warn!("put_link failed");
-                }
+        let ld = match deserialize::<InterfaceLinkDefinition>(&msg.payload) {
+            Ok(ld) => ld,
+            Err(e) => {
+                error!(error = %e, "received invalid link def data on message");
+                return;
             }
-            Err(err) => {
-                error!(%err, "received invalid link def data on message");
+        };
+
+        let span = tracing::Span::current();
+        span.record("source_id", &tracing::field::display(&ld.source_id));
+        span.record("target", &tracing::field::display(&ld.target));
+        span.record("wit_namespace", &tracing::field::display(&ld.wit_namespace));
+        span.record("wit_package", &tracing::field::display(&ld.wit_package));
+        span.record(
+            "wit_interfaces",
+            &tracing::field::display(&ld.interfaces.join(",")),
+        );
+        span.record("link_name", &tracing::field::display(&ld.name));
+        // If the link has already been put, return early
+        if self.is_linked(&ld.source_id).await {
+            warn!("Ignoring duplicate link put");
+            return;
+        }
+
+        // Get the config bundles that apply to the named configs on the link
+        debug!("retrieving named config for link");
+        let config = self
+            .host_data
+            .configs
+            .get(&ld.name)
+            .cloned()
+            .unwrap_or_default();
+
+        info!("Linking actor with provider");
+        match provider
+            .receive_link_config_as_target((&ld.source_id, &ld.target, &ld.name, &config))
+            .await
+        {
+            Ok(()) => self.put_link(ld).await,
+            Err(e) => {
+                warn!(error = %e, "receive_link_config_as_target() failed");
```

Comment:
I think here we may need to actually check to see if the `source_id` == `provider_key` in order to tell if this is a source or target config

Reply:
Yeah I did this in the other spot, will do here as well